### ISSUE TITLE
Finding dimensions of arbitrary HTML

### ISSUE THREAD

**Issue by scripting** [cburgmer/rasterizeHTML.js#170]
I have an arbitrary string of styled text, that I want to render into a canvas. The width is fixed at 400 pixels. I want to determine the height of the rendering, so I can create a canvas, in JS, to contain it, with proper margins.

I tried doing two passes, one to generate the image of the text, hoping the height would be a function of the height of the rendered text, but it's fixed.

I'm looking for an elegant solution, failing that an egregious hack is okay too. ;-)

Thanks in adavance, and I apologize in advance if this question has been asked previously or is documented.

**Comment by cburgmer** [comment]
Sounds like a sane requirement.

If you provide a fixed width, and then a fixed height of only say 10px, does that give you the dynamic height as needed? You'll definitely always have to go through the `.then()` handler to obtain the image first, but that should get the job done. 

More specifically, the width/height you provide is used to initially size the viewport of the renderer. After the rendering we read back the final content pane's size and return those dimensions for the image.

**Comment by scripting** [comment]
1. Yes I am using the .then () branch to determine the image dimensions.

2. I tried setting the height at 10px on the first pass. 

3. Amazed that it worked. rasterizeHtml, in the then () handle shows the img.height at 290. Which is the exact height of the rendered text.

4. So your advice worked!

5. What's amazing to me is that I didn't try this. I think I tried everything else. 

Anyway, thanks for the answer, and of course thanks for the tool. It's making some interesting stuff possible. Hope to show it soon.
